### ISSUE TITLE
Adding albums for new artists to the wishlist

### ISSUE THREAD

**Issue by Phyks** [ampache/ampache#1109]
Hi,

I have just discovered the wishlist feature in Ampache, which is really cool. I could add missing albums from artists already in my library to the wishlist, to keep track of it.

However, I could not find any way to add albums from an artist which is not currently in my library (if I do not yet have any CD of him for instance).

Is there a way to do it? Would be nice to see it in a future version, else!

Thanks

**Comment by Afterster** [comment]
Yes there is a way. You can do it by browsing similar artists, or searching on Missing Artists (top search of each page).

**Comment by Phyks** [comment]
Ok, I missed it. 

Some remarks however:
* First, would it be possible to have some extra disambiguation in the artist names (_à la_ Musicbrainz). If I am looking for "Ghost", there is not less than 17 different matching groups, and it is really difficult to know on which I want to click.

* Is there a way to add an unknown artist to the wishlist? I would like to have a centralized management of my wishes, but some artists are unknown to Ampache (Beat'O'Cue for instance). Having a form and manually filling artist and album would be really cool.

* I added a lot of albums to my wishlist, and they seem to all be stored in the db. However, only a fraction of them is shown on the wishlist page. It is as if there was a pagination mechanism, but the page numbers were masked.

* Small typo, https://github.com/ampache/ampache/blob/620ca4056f54567b26327e3d0bdc5a6f19f5547b/locale/fr_FR/LC_MESSAGES/messages.po#L4032 should be `artistes manquants` (without extra `e`).

Thanks

**Comment by Afterster** [comment]
For 1) and 2), missing/wanted feature is totally based on MusicBrainz so there is no possibility for now.
3) looks like a missing paging here.
4) No possibility to share the whishlist. But administrators see all wanted albums for all users on his view.
5) Thanks, I will take the opportunity to point you to [Transifex page](https://www.transifex.com/ampache/ampache/) if you want to fix/contribute directly in the future

**Comment by Phyks** [comment]
For 1) I don't get it. If I go to https://musicbrainz.org/search?query=ghost&type=artist&method=indexed there are some disambiguations listed near the names (see "(UK garage)"). I do not know much about the backend infrastructure, but I believe they are accessible through the API and could be displayed in Ampache.

For 2), ok. This would be a great feature!

For 3), indeed, but I do not see such things in the rendered HTML. =(

For 4), ok. Same, would be a great feature.

For 5), thanks! I will go to Transifex and see if I can give a hand if I manage to find some time, to handle the lacking strings in French translation.

**Comment by Afterster** [comment]
Right :smile:. 1) and 3) should be an improvement, as 2) and 4) but it would require more changes.

**Comment by Phyks** [comment]
Also, it seems that loading of albums on a missing artist page is deferred, but there are no loading indicators on the missing artist page, which is not very good UX :)

**Comment by Phyks** [comment]
@Afterster Another bug with the "missing artist" feature.

If I search for "Shake shake go" in "missing artist", I can find "Shake shake go" (which is ok, since it exists on Musicbrainz), but Ampache does not find any associated album.

However, there are some albums on Musicbrainz, see https://musicbrainz.org/artist/5840b983-11fb-47a8-85a7-8340518772bb.

My bet is that Ampache is looking for "Albums" and discard "EPs", which is not really cool. :/

**Comment by Afterster** [comment]
@Phyks that's right but you can configure it. See `wanted_types` on ampache.cfg.php

**Comment by Phyks** [comment]
Indeed, it did the work. Maybe a safer default concerning this point would be to add more types by default to this config file?

**Comment by Phyks** [comment]
Also, just noted it, but once the wished album is imported in the library, it does not automatically get removed from the wishlist :)

**Comment by Phyks** [comment]
Concerning 

> 3) looks like a missing paging here.

Actually, looking at the code, I found out that this is due to the fact that the same offset restriction is set across all ampache. Then, by default, the wishlist is truncated to 50 items, and there is no pagination on top of it. Changing the global offset should fix it, but this is some weird behaviour in my opinion (plus it may be affecting other templates as well).

**Comment by Phyks** [comment]
Just suming it up with a checkboxes list:
- [ ] Fetch the disambiguation in the artist names from Musicbrainz and display it. If I am looking for "Ghost", there is not less than 17 different matching groups, and it is really difficult to know on which I want to click.

- [ ] Add a way to add an artist based on its name, which is not in Musicbrainz db (field to type artist name as plaintext).

- [ ] There is a bug with the pagination mechanism on the wishlist page.

- [ ] Wishlist is reaaaaallly slow to load.

- [ ] Add a way to share the wishlist with others.

- [ ] What about EPs which are disregarded by default?

- [ ] Add a loading indicator on the missing artist page.

- [ ] Fix typo in French translation in Transiflex.
